indexing-service/src/test/java/org/apache/druid/indexing/overlord/hrtr/HttpRemoteTaskRunnerTest.java

```diff
@@ -1198,6 +1216,208 @@ public void testTaskAddedOrUpdated3()
+
+    Capture<Request> capturedRequest = getHttpClientRequestCapture(HttpResponseStatus.INTERNAL_SERVER_ERROR, "");
+
+    try {
+      taskRunner.disableWorker("host:1234");
+      Assert.fail("Should raise RE exception!");
+    }
+    catch (RE re) {
+    }
+
+    Assert.assertEquals(HttpMethod.POST, capturedRequest.getValue().getMethod());
+    Assert.assertEquals(url, capturedRequest.getValue().getUrl());
+
+    EasyMock.verify(httpClient);
+  }
+
+  @Test(expected = RE.class)
+  public void testDisableWorkerWhenWorkerNotExists()
+  {
+    HttpRemoteTaskRunner taskRunner = createTaskRunnerForEnableDisableWorkerTests();
+
+    taskRunner.disableWorker("unkown-host:1234");
+
+    EasyMock.replay(httpClient);
+    EasyMock.verify(httpClient);
+  }
+
+  @Test
+  public void testEnableWorker() throws Exception
+  {
+    HttpRemoteTaskRunner taskRunner = createTaskRunnerForEnableDisableWorkerTests();
+
+    final URL url = new URL("http://host:1234/druid/worker/v1/enable");
+    String workerResponse = "{\"host:1234\":\"enabled\"}";
+
+    Capture<Request> capturedRequest = getHttpClientRequestCapture(HttpResponseStatus.OK, workerResponse);
+
+    taskRunner.enableWorker("host:1234");
+
+    Assert.assertEquals(HttpMethod.POST, capturedRequest.getValue().getMethod());
+    Assert.assertEquals(url, capturedRequest.getValue().getUrl());
+
+    EasyMock.verify(httpClient);
+  }
+
+  @Test
+  public void testEnableWorkerWhenWorkerRaisesError() throws Exception
+  {
+    HttpRemoteTaskRunner taskRunner = createTaskRunnerForEnableDisableWorkerTests();
+    final URL url = new URL("http://host:1234/druid/worker/v1/enable");
+
+    Capture<Request> capturedRequest = getHttpClientRequestCapture(HttpResponseStatus.INTERNAL_SERVER_ERROR, "");
+
+    try {
+      taskRunner.enableWorker("host:1234");
+      Assert.fail("Should raise RE exception!");
+    }
+    catch (RE re) {
+    }
+
+    Assert.assertEquals(HttpMethod.POST, capturedRequest.getValue().getMethod());
+    Assert.assertEquals(url, capturedRequest.getValue().getUrl());
+
+    EasyMock.verify(httpClient);
+  }
+
+  @Test(expected = RE.class)
+  public void testEnableWorkerWhenWorkerNotExists()
+  {
+    HttpRemoteTaskRunner taskRunner = createTaskRunnerForEnableDisableWorkerTests();
+
+    taskRunner.enableWorker("unkown-host:1234");
+
+    EasyMock.replay(httpClient);
+    EasyMock.verify(httpClient);
+  }
+
+  private HttpRemoteTaskRunner createTaskRunnerForEnableDisableWorkerTests()
+  {
+    TestDruidNodeDiscovery druidNodeDiscovery = new TestDruidNodeDiscovery();
+    DruidNodeDiscoveryProvider druidNodeDiscoveryProvider = EasyMock.createMock(DruidNodeDiscoveryProvider.class);
+    EasyMock.expect(druidNodeDiscoveryProvider.getForNodeType(NodeType.MIDDLE_MANAGER))
+            .andReturn(druidNodeDiscovery);
+    EasyMock.replay(druidNodeDiscoveryProvider);
+
+    Map<String, CustomFunction> workerHolders = new ConcurrentHashMap<>();
```

Comment:
This line is causing a teamcity failure, I think you should likely just assign to a `ConcurrentHashMap`?
![Screen Shot 2019-05-15 at 6 01 00 PM](https://user-images.githubusercontent.com/1577461/57819035-6b63f700-773b-11e9-8a52-9e6d69ca2676.png)

Reply:
Fixed.